types/block.go

```diff
@@ -76,19 +79,44 @@ func (roots NmtRoots) Bytes() [][]byte {
 	return res
 }
 
-func NmtRootsFromBytes(in [][]byte) NmtRoots {
-	roots := make([]namespace.IntervalDigest, len(in))
+func NmtRootsFromBytes(in [][]byte) (roots NmtRoots, err error) {
+	roots = make([]namespace.IntervalDigest, len(in))
 	for i := 0; i < len(in); i++ {
-		roots[i] = namespace.IntervalDigestFromBytes(NamespaceSize, in[i])
+		roots[i], err = namespace.IntervalDigestFromBytes(NamespaceSize, in[i])
+		if err != nil {
+			return
+		}
 	}
-	return roots
+	return
 }
 
-// Hash computes the root of the row and column roots
+// String returns hex representation of merkle hash of the DAHeader.
+func (dah *DataAvailabilityHeader) String() string {
+	if dah == nil {
+		return "<nil DAHeader>"
+	}
+	return strings.ToUpper(hex.EncodeToString(dah.Hash()))
```

Comment:
What motivates the hex encoding here again? Isn't a simple fmt.SPrintf("%x\n", dah.Hash()) similarly useful for debugging?

Reply:
Why again?

Reply:
Didn't know about the `%x` verb actually, so will change that